src/internal/transaction.ts

```diff
@@ -105,6 +105,7 @@ export async function waitForTransaction(args: {
   const timeoutSecs = options?.timeoutSecs ?? DEFAULT_TXN_TIMEOUT_SEC;
   const checkSuccess = options?.checkSuccess ?? true;
   const indexerVersionCheck = options?.indexerVersionCheck ?? true;
```

Comment:
I wonder if we should update this and default to `false`. I feel like, only dev that relies on indexer API will probably want. But if they are only sending request to full node api, they probably want to have this off.

Thought? cc: @0xmaayan @gregnazario

Reply:
hmm most of the sdk queries use indexer, so if a sdk developer uses the sdk to submit transactions and then fetch data, they/sdk should wait for indexer.

Reply:
> Only annoying thing is... if it's by default on and the indexer is delayed / broken, the dapps will completely not work.
> 
> It's a strange tradeoff based on what your dapp does.

Yeah, I am thinking about this trade off as well. Because I would hate to sdk to be broken b/c of indexer. Maybe we should just always return, but have error field saying that the indexer failed to sync?

For example:
const result = aptos.waitForTransaction();

and result looks like following:
{
  indexerSyncError: "timeout"
  ...
}


Or we just relies on the developer to wrap `waitForTransaction` in a try-catch